Authentication flow

Derek-Chu opened the pull request:
Changed screens to use react-native-paper. Added a forgot password screen to allow users to reset password. Got rid of Edit Profile button on home screen and also changed Sign Out button to redirect to SignInScreen instead of SignOutScreen.

Derek-Chu committed:
Added React-Native Paper

Derek-Chu committed:
Added password reset

Got rid of Edit Profile button on HomeScreen

Derek-Chu committed:
Changed EditProfile to use react-native-paper

Also changed sign out button on home screen to navigate to SignInScreen instead of SignUpScreen

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -54,7 +57,7 @@ class EditProfileScreen extends React.Component {
       <View style={styles.container}>
-        <Text>Edit Profile</Text>
+        <Title>Edit Profile</Title>
         <TextInput
```
Focus the cursor on the next TextInput when the user hits enter (see SignInScreen)

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -71,7 +74,13 @@ class EditProfileScreen extends React.Component {
           onChangeText={(text) => this.setState({ newPassword: text })}
           value={newPassword}
```
Add another prop to this TextInput so that hitting enter calls onChangePasswordPress

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -71,7 +74,13 @@ class EditProfileScreen extends React.Component {
+          Change Password
+        </Button>
 
         <TextInput
```
Same as above

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -81,7 +90,13 @@ class EditProfileScreen extends React.Component {
           value={newEmail}
         />
-        <Button title="Change Email" onPress={this.onChangeEmailPress} />
+        <Button
```
Same as above

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -54,7 +57,7 @@ class EditProfileScreen extends React.Component {
     const { currentPassword, newPassword, newEmail } = this.state;
```
Add a loading state that displays an ActivityIndicator and disables the input fields when loading

vfcheung commented on src/screens/Home/HomeScreen.js:
```diff
@@ -52,14 +52,19 @@ class HomeScreen extends React.Component {
+          style={styles.button}
+          mode="contained"
+          onPress={this.handleSignOut}
+        >
```
Add loading state and ActivityIndicator

Derek-Chu committed:
QOL updates to EditProfileScreen

Incorporated feeback from @vfcheung (making enter buttons for all TextInputs functionally intuitive, adding in loading state for email and password change with relevant ActivityIndicators). Added autofocus to current password input field and cleared inputs after submitting password/email change request.

vfcheung commented on src/screens/Home/HomeScreen.js:
```diff
@@ -34,15 +35,14 @@ class HomeScreen extends React.Component {
       .signOut()
       .then(() => {
-        navigation.navigate('SignUpScreen');
+        navigation.navigate('SignInScreen');
```
Can you navigate to `Auth`?

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -21,26 +26,58 @@ class EditProfileScreen extends React.Component {
+          })
+          .catch((error) => {
+            Alert.alert(error.message);
+            this.clearFields();
```
Can you use `this.setState(INITIAL_STATE)` to clear the fields?

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -21,26 +26,58 @@ class EditProfileScreen extends React.Component {
-      .catch((error) => Alert.alert(error.message));
+      .catch((error) => {
+        Alert.alert(error.message);
+        this.clearFields();
```
Same

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -21,26 +26,58 @@ class EditProfileScreen extends React.Component {
-          .catch((error) => Alert.alert(error.message));
+          .then(() => {
+            Alert.alert('Email was changed successfully!');
+            this.clearFields();
```
Same

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -21,26 +26,58 @@ class EditProfileScreen extends React.Component {
+          })
+          .catch((error) => {
+            Alert.alert(error.message);
+            this.clearFields();
```
Same

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -21,26 +26,58 @@ class EditProfileScreen extends React.Component {
-      .catch((error) => Alert.alert(error.message));
+      .catch((error) => {
+        Alert.alert(error.message);
+        this.clearFields();
```
Same

vfcheung commented on src/screens/Profile/EditProfileScreen.js:
```diff
@@ -21,26 +26,58 @@ class EditProfileScreen extends React.Component {
+  clearFields() {
+    this.setState({ currentPassword: '' });
+    this.setState({ newPassword: '' });
+    this.setState({ newEmail: '' });
```
Probably don't need

Derek-Chu committed:
Small refactoring
